src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventProvider.cs

```diff
@@ -628,7 +628,8 @@ private static int IndexOfSessionInList(List<SessionInfo>? sessions, int etwSess
                 if (filterData->Ptr != 0 && 0 < filterData->Size && filterData->Size <= 100*1024)
                 {
                     data = new byte[filterData->Size];
-                    Marshal.Copy((IntPtr)filterData->Ptr, data, 0, data.Length);
+                    IntPtr ptr = new IntPtr(IntPtr.Size == 4 ? (int)filterData->Ptr : filterData->Ptr);
+                    Marshal.Copy(ptr, data, 0, data.Length);
```

Comment:
It would be simpler to do just:

`Marshal.Copy((IntPtr)(void*)filterData->Ptr, data, 0, data.Length);`

Reply:
Thank you!